hbase-server/src/test/java/org/apache/hadoop/hbase/TestClusterPortAssignment.java

```diff
@@ -44,16 +46,20 @@
   @Test
   public void testClusterPortAssignment() throws Exception {
     boolean retry = false;
+    int retryCount = 0;
     do {
       int masterPort =  HBaseTestingUtility.randomFreePort();
       int masterInfoPort =  HBaseTestingUtility.randomFreePort();
       int rsPort =  HBaseTestingUtility.randomFreePort();
       int rsInfoPort =  HBaseTestingUtility.randomFreePort();
       TEST_UTIL.getConfiguration().setBoolean(LocalHBaseCluster.ASSIGN_RANDOM_PORTS, false);
+      TEST_UTIL.getConfiguration().setBoolean(HConstants.REGIONSERVER_INFO_PORT_AUTO, false);
+      TEST_UTIL.getConfiguration().setBoolean("fs.hdfs.impl.disable.cache", true);
```

Comment:
interesting, I'll have to look into this feature. I'm not sure why we need this. Which error do we see if we don't have this? I don't see any other test using it, but maybe none of them try to quickly restart?

Reply:
Maybe. The restart failure happens here: https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ShutdownHook.java#L207

I guess previous shutdown in the same jvm has left anything for hdfs to do in cache.